src/datanode/src/sql/create.rs

```diff
@@ -135,6 +135,35 @@ impl SqlHandler {
             .map(|(k, v)| (v, k))
             .collect::<HashMap<_, _>>();
 
+        let pk_map = stmt
+            .columns
+            .iter()
+            .filter(|col| {
+                col.options
+                    .iter()
+                    .any(|options| match options.option.clone() {
+                        ColumnOption::Unique { is_primary } => is_primary,
+                        _ => false,
+                    })
+            })
+            .enumerate()
+            .map(|(_, col)| col.name.value.clone())
+            .collect::<Vec<_>>();
+
+        ensure!(
+            pk_map.len() < 2,
+            InvalidPrimaryKeySnafu {
+                msg: "Multiple definitions of primary key found"
+            }
+        );
+
+        if pk_map.first().is_some() {
+            let pk = pk_map.first().unwrap();
+            primary_keys.push(*col_map.get(&pk.clone()).context(KeyColumnNotFoundSnafu {
+                name: pk.to_string(),
+            })?);
```

Comment:
Two points:
- Clone in `&pk.clone()` should be avoidable. Taking reference from a cloned value looks odd.
- I think it is safe to use unwrap here. I also add a safety note in the comment.
```suggestion
            // # Safety: Both pk_map and col_map are collected from stmt.columns
            primary_keys.push(*col_map.get(pk).unwrap());
```

Reply:
fixed